viewing_party/party.py

```diff
@@ -1,23 +1,166 @@
+    genre_list = []
+    genre = ""
+
+    # If the value of "watched" is an empty list, get_most_watched_genre should return None.
+    if user_data["watched"] == []:
+        return None
 
+    for movie in user_data["watched"]: #this giving each movie dictionary inside the watched list 
+        genre = movie['genre'] #accessing the movie genre and assigning to the genre 
+        genre_list.append(genre) #each genre will be added to the genre_list # the congregate of all genres
 
+
+    favorite_genre = max(set(genre_list), key= genre_list.count) 
+    return favorite_genre
+        
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    user_not_watched_movies = []
+    movies_watched_by_friends = _get_movies_watched_by_friends(user_data) 
+
+    for movie in user_data["watched"]:
+        if movie not in movies_watched_by_friends:
+            user_not_watched_movies.append(movie)
+    return user_not_watched_movies
+
+
+def get_friends_unique_watched(user_data):
+    recommended_movies = []
+    
+    movies_watched_by_friends = []
+    movies_watched_by_friends = _get_movies_watched_by_friends(user_data) 
+    user_watched_movies = user_data["watched"] 
+
+    for friend_movie in movies_watched_by_friends: #user_data = "friends": [{}]
+        if friend_movie not in user_watched_movies:
+            recommended_movies.append(friend_movie)
+    
+    return recommended_movies
+
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    recommended_movies = []
+    recommended_movies_subscription = []
+
+    user_watched_movies = user_data["watched"] 
+    user_subscriptions = user_data["subscriptions"]
+    movies_watched_by_friends = _get_movies_watched_by_friends(user_data)
 
+    if user_data["friends"] == []:
+        return len(recommended_movies)
+
+    for friend_movie in movies_watched_by_friends: 
+        if friend_movie not in user_watched_movies:
+            recommended_movies.append(friend_movie)
+
+    for movie in recommended_movies:
+        if movie["host"] in user_subscriptions:
+            pprint(recommended_movies_subscription.append(movie))
+
+    return recommended_movies_subscription
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    list_rec_movies_genre = []
+
+    popular_genre = get_most_watched_genre(user_data)
+
+    friend_movies = _get_movies_watched_by_friends(user_data)
+
+    for friend_movie in friend_movies:
+        if friend_movie["genre"] == popular_genre:
+            if friend_movie not in user_data["watched"]:
+                list_rec_movies_genre.append(friend_movie)
+    print(list_rec_movies_genre)

```

Comment:
It's usually not a good idea to leave print statements in committed code because they're often for debug purposes. It's good to get in the habit of re-reading your PRs to make sure they don't contain things like this.